controlplane/pkg/monitor_crossconnect_server/monitor_crossconnect_client.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2018 Cisco and/or its affiliates.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package monitor_crossconnect_server
+
+import (
+	"context"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/ligato/networkservicemesh/controlplane/pkg/apis/crossconnect"
+	"github.com/ligato/networkservicemesh/controlplane/pkg/apis/registry"
+	"github.com/ligato/networkservicemesh/controlplane/pkg/model"
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	"net"
+	"time"
+)
+
+type nsmMonitorCrossConnectClient struct {
+	monitor    MonitorCrossConnectServer // All connections is here
+	model      model.Model
+	dataplanes map[string]*dataplaneCrossConnectInfo
+}
+
+type dataplaneCrossConnectInfo struct {
+	crossConnects map[string]*crossconnect.CrossConnect
+}
+
+func (client *nsmMonitorCrossConnectClient) EndpointAdded(endpoint *registry.NetworkServiceEndpoint) {
+}
+
+func (client *nsmMonitorCrossConnectClient) EndpointDeleted(endpoint *registry.NetworkServiceEndpoint) {
+}
+
+func (client *nsmMonitorCrossConnectClient) DataplaneAdded(dataplane *model.Dataplane) {
+	info := &dataplaneCrossConnectInfo{
+		crossConnects: make(map[string]*crossconnect.CrossConnect),
```

Comment:
The MonitorCrossConnectServer already encapsulates within itself a store for CrossConnects.  I'm curious why we are storing them a second time here :)

Reply:
We need this to be able to remove in case of dataplane will be removed, since in connections itself there is no information about how was added/updated it. Also it is a map of references to same cross connection objects, so should not eat so many memory.